docs/notebooks/5_warminglevels.ipynb

```diff
@@ -171,227 +166,219 @@
     ")\n",
-    "\n",
-    "for id_wl, ds_wl in dict_wl.items():\n",
-    "    # for this simulation, find the accompanying reference horizon\n",
-    "    level = ds_wl.attrs[\"cat:processing_level\"]\n",
-    "    id_hor = id_wl.replace(level, f\"clim{ref_period}\")\n",
-    "    ds_hor = dict_hor[id_hor]\n",
-    "\n",
-    "    # concat warming level and reference\n",
-    "    ds_concat = xr.concat([ds_wl, ds_hor], dim=\"horizon\")\n",
-    "\n",
-    "    # compute delta\n",
-    "    ds_delta = xs.aggregate.compute_deltas(\n",
-    "        ds=ds_concat, reference_horizon=ref_period, to_level=f\"delta-{level}\"\n",
-    "    )\n",
-    "\n",
-    "    # Save\n",
-    "    filename = str(\n",
-    "        output_folder\n",
-    "        / f\"wl_{ds_delta.attrs['cat:id']}.{ds_delta.attrs['cat:domain']}.{ds_delta.attrs['cat:processing_level']}.{ds_delta.attrs['cat:frequency']}.zarr\"\n",
-    "    )\n",
-    "    xs.save_to_zarr(ds_delta, filename, mode=\"o\")\n",
-    "    pcat.update_from_ds(ds=ds_delta, path=filename, info_dict={\"format\": \"zarr\"})"
+    "display(weights)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "7f96ae9b",
-   "metadata": {},
+   "id": "dd9b6a71-0ee8-4c3e-9962-d18e4b19a1fa",
+   "metadata": {
+    "tags": []
+   },
    "outputs": [],
    "source": [
-    "display(ds_delta)"
+    "# When split_experiments=False, all realisations of NorESM2-MM are given the same weight\n",
+    "weights = xs.ensembles.generate_weights(\n",
+    "    datasets=datasets, independence_level=\"model\", split_experiments=False, skipna=False\n",
+    ")\n",
+    "display(weights)"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "d0e70707",
+   "id": "b640fa35-d68d-4dbd-83cd-f1a1b3aa974c",
    "metadata": {},
    "source": [
-    "## Ensemble statistics\n",
-    "\n",
-    "Even more than with time-based horizons, the first step of ensemble statistics should be to generate the weights. Indeed, if a model has 3 experiments reaching a given warming level, we want it to have the same weight as a model with only 2 experiments reaching that warming.\n",
-    "\n",
-    "<div class=\"alert alert-warning\"> <b>WARNING</b>\n",
-    "    \n",
-    "`xs.ensembles.generate_weights` is currently purely based on metadata, and thus cannot distinguish subtleties about which realization reaches which warming level if multiple experiments are concatenated together before passing them to the function. The results are likely to be wrong, which is why each warming level needs to be computed individually.\n",
-    "</div>\n",
-    "\n",
-    "Next, the weights and the datasets can be passed to `xs.ensemble_stats` to calculate the ensemble statistics."
+    "Next, the weights and the datasets can be passed to `xs.ensemble_stats` to calculate the ensemble statistics. Note that if `split_experiments=True` is used, the datasets should be split before being sent to `ensemble_statistics`."
```

Comment:
You bring out a going counterpoint that even though we can now more easily weight the experiments separately, it's not really useful. For now, I've removed text related to this from the notebooks, and `split_experiments == False` is the default anyway.

Two options:

1. I change it so that an equal weight is given to all experiments. i.e. if SSP585 has 30 simulations and SSP245 only has 5, the total weight is still 1 each. This would allow for some fancy ensemble statistics that are not possible if you compute each experiment separately.
2. I keep it as it is, it just won't be too useful.

Reply:
I don't see when we would want 1... so I would vote for 2 !

Reply:
A case for #1 (which is kinda what I had in mind when designing the option) would be if you want to make statistics on the full scope of possible climate change. @sarahclaude will need to do that for her project (although in her case, we would need to allow the weight between experiments to differ from 1).

Reply:
Yes it will be useful to be able to associate weights to experiement in an ensemble! I am not using it directly in ensemble_stats.